### PR TITLE
bug/many2one_relationship_dropdown

### PR DESCRIPTION
**Pull request by tristanblc**
Hi,
I created new controller ( PublicUsersControllers) who return only activated users. 
This controller have route "/api/users".

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/dotnetcore/src/Project.Application/Queries/User/UserGetAllPublicUsersQueryHandler.cs.ejs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        _userManager = userManager;
+        _mapper = mapper;
+    }
+
+    public async Task<(IHeaderDictionary, IEnumerable<UserDto>)> Handle(UserGetAllPublicUsersQuery request, CancellationToken cancellationToken)
+    {
+        <%_ if (databaseType === 'mongodb') { _%>
+        var page = await Task.FromResult((_userManager.Users as IMongoQueryable<User>)
+            .Include(it => it.UserRoles)
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project.Application/Queries/User/UserGetAllPublicUsersQueryHandler.cs.ejs:55]: user roles and role is not necessary

File: generators/server/templates/dotnetcore/src/Project.Application/Queries/User/UserGetAllPublicUsersQueryHandler.cs.ejs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            .Where(it => it.Activated == true && !string.IsNullOrEmpty(it.Id))
+            .UsePageable(request.Page));
+        <%_ } else { _%>
+        var page = await _userManager.Users
+            .Include(it => it.UserRoles)
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project.Application/Queries/User/UserGetAllPublicUsersQueryHandler.cs.ejs:60]: user roles and role is not necessary

File: generators/server/templates/dotnetcore/src/Project.Domain.Services/UserService.cs.ejs [hunk 3]
```diff
@@ -91,6 +91,22 @@ public class UserService : IUserService
         <%_ } _%>
     }
 
+    public async Task<IPage<User>> GetAllPublicUsers(IPageable pageable)
+    {
+        <%_ if (databaseType === 'mongodb') { _%>
+        IPage<User> users = await (_userManager.Users as IMongoQueryable<User>)
+                 .Include(it => it.UserRoles)
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project.Domain.Services/UserService.cs.ejs:98]: user roles and role is not necessary

File: generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+using <%= namespace %>.Infrastructure.Web.Rest.Utilities;
+<%_ if (cqrsEnabled) { _%>
+using <%= namespace %>.Application.Queries;
+using <%= namespace %>.Application.Commands;
+<%_ } _%>
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.EntityFrameworkCore;
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs:43]: there are multiple unused using

File: generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        _userManager = userManager;
+        _userService = userService;
+        _mailService = mailService;
+        _mapper = mapper;
+    }
+    <%_ } _%>
+
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<User>>> GetAllPublicUsers(IPageable pageable)
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs:81]: return an USERDTO here

File: generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+[Route("api/users")]
+[ApiController]
+public class PublicUsersController : ControllerBase
+{
+    private readonly ILogger<UsersController> _log;
+    <%_ if (cqrsEnabled) { _%>
+    private readonly IMediator _mediator;
+    <%_ } else { _%>
+    private readonly IMapper _mapper;
+    private readonly IMailService _mailService;
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs:56]: Can you remove unused fields (mailservice and user manager)

**Review by nicolas63 — CHANGES_REQUESTED** [review]

File: generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs (hunk 6 above)
Comment (tristanblc) [reply generators/server/templates/dotnetcore/src/Project/Controllers/PublicUsersController.cs.ejs:56]: Yes, I can

**Review by nicolas63 — APPROVED** [review]
Looks good to me, thanks for your contribution !

**Pull request merged**